lib/auth/appauthconfig/appauthconfigv1/service_test.go

```diff
@@ -0,0 +1,315 @@
+				require.NoError(t, err)
+				_, err = clt.CreateAppAuthConfig(ctx, newAppAuthConfigJWT())
+				require.NoError(t, err)
+				resp, err := clt.ServiceUnderTest.ListAppAuthConfigs(ctx, &appauthconfigv1.ListAppAuthConfigsRequest{})
+				if err == nil {
+					require.NotNil(t, resp)
+					require.Len(t, resp.Configs, 2, "expected 2 inserted")
+				}
+				return err
+			},
+			requiredAccessRules: []types.Rule{appAuthConfigRule(types.VerbRead, types.VerbList)},
+		},
+		// Update
+		{
+			name: "UpdateAppAuthConfig",
+			actionFn: func(t *testing.T, ctx context.Context, clt testClient) error {
+				cfg, err := clt.CreateAppAuthConfig(ctx, newAppAuthConfigJWT())
+				require.NoError(t, err)
+				firstRev := cfg.Metadata.Revision
+				cfg, err = clt.ServiceUnderTest.UpdateAppAuthConfig(ctx, &appauthconfigv1.UpdateAppAuthConfigRequest{
+					Config: cfg,
+				})
+				if err == nil {
+					require.NotNil(t, cfg, "the updated resource should be returned")
+					require.NotEqual(t, firstRev, cfg.Metadata.Revision, "the resource should have been updated")
+				}
+				return err
+			},
+			requiredAccessRules: []types.Rule{appAuthConfigRule(types.VerbUpdate)},
+			requireEvent:        &apievents.AppAuthConfigUpdate{},
+		},
+		{
+			name: "UpsertAppAuthConfig",
+			actionFn: func(t *testing.T, ctx context.Context, clt testClient) error {
+				cfg := newAppAuthConfigJWT()
+				_, err := clt.ServiceUnderTest.UpsertAppAuthConfig(ctx, &appauthconfigv1.UpsertAppAuthConfigRequest{
+					Config: cfg,
+				})
+				return err
+			},
+			requiredAccessRules: []types.Rule{appAuthConfigRule(types.VerbCreate, types.VerbUpdate)},
+			requireEvent:        &apievents.AppAuthConfigCreate{},
+		},
+		// Delete
+		{
+			name: "DeleteAppAuthConfig",
+			actionFn: func(t *testing.T, ctx context.Context, clt testClient) error {
+				cfg, err := clt.CreateAppAuthConfig(ctx, newAppAuthConfigJWT())
+				require.NoError(t, err)
+				_, err = clt.ServiceUnderTest.DeleteAppAuthConfig(ctx, &appauthconfigv1.DeleteAppAuthConfigRequest{
+					Name: cfg.Metadata.GetName(),
+				})
+				return err
+			},
+			requiredAccessRules: []types.Rule{appAuthConfigRule(types.VerbDelete)},
+			requireEvent:        &apievents.AppAuthConfigDelete{},
+		},
+	}
+	for _, test := range tests {
+		test.run(t)
+	}
+}
+
+func appAuthConfigRule(verbs ...string) types.Rule {
+	return types.Rule{
+		Resources: []string{types.KindAppAuthConfig},
+		Verbs:     verbs,
+	}
+}
+
+type accessTest struct {
+	name     string
+	actionFn func(t *testing.T, ctx context.Context, clt testClient) error
+	// requiredAccessRules are rules that are expected to be allowed and not
+	// denied to perform actionFn without an access error.
+	requiredAccessRules []types.Rule
+	// requireEvent must be emitted if the action succeeds.
+	requireEvent apievents.AuditEvent
+}
+
+func (c *accessTest) setup(t *testing.T, specs ...types.RoleSpecV6) (context.Context, testClient) {
+	t.Helper()
+	clt := newService(t, context.Background())
```

Comment:
nit: Prefer using t.Context() in tests.
```suggestion
	clt := newService(t, t.Context())
```